tests/protocol/test_hostplugin.py

```diff
@@ -536,6 +538,44 @@ def test_validate_page_blobs(self):
                         test_goal_state,
                         exp_method, exp_url, exp_data)
 
+    def test_validate_http_request_when_uploading_logs(self):
+        with mock_wire_protocol(DATA_FILE) as protocol:
+            test_goal_state = protocol.client._goal_state
+            correlation_id = str(uuid.uuid4())
+
+            expected_url = hostplugin.URI_FORMAT_PUT_LOG.format(wireserver_url, hostplugin.HOST_PLUGIN_PORT)
+            expected_headers = {'x-ms-version': '2015-09-01',
+                                "x-ms-containerid": test_goal_state.container_id,
+                                "x-ms-vmagentlog-deploymentid": test_goal_state.role_config_name.split(".")[0],
+                                "x-ms-client-name": AGENT_NAME,
+                                "x-ms-client-version": AGENT_VERSION,
+                                "x-ms-client-correlationid": correlation_id}
+
+            host_client = wire.HostPluginProtocol(wireserver_url,
+                                                  test_goal_state.container_id,
+                                                  test_goal_state.role_config_name)
+
+            self.assertFalse(host_client.is_initialized, "Host plugin should not be initialized!")
+
+            with patch.object(restutil, "http_request") as patch_http:
```

Comment:
try using the http_handlers in mock_wire_protocol instead of mocking http_request